### PR TITLE
Add "lesson 12" example about point lighting

### PR DESCRIPTION
**Pull request by jianhuang01**
New "lesson 12" example focusing on point lighting on a textured sphere and a textured cube, lighting is based on per-vertex shading.

### CODE REVIEW

File: examples/lessons/12/app.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  },
+  onRender: ({
+    gl, tick, aspect, moon, cube
+  }) => {
+    // Update Camera Position
+    const eyePos = [0, 0, 20];
+
+    gl.clear(GL.COLOR_BUFFER_BIT | GL.DEPTH_BUFFER_BIT);
+
+    let uMoonMMatrix = new Matrix4()
```
Comment (ibgreen) [inline examples/lessons/12/app.js]: Doesn't this just set `uMoonMatrix` to `moonRotation.matrix`?

Can't just pass `moonRotation.matrix` to `uMoonMatrix` below in set uniforms?

File: examples/lessons/12/app.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        });
+        return {moon, cube};
+      });
+    });
+  },
+  onRender: ({
+    gl, tick, aspect, moon, cube
+  }) => {
+    // Update Camera Position
+    const eyePos = [0, 0, 20];
```
Comment (ibgreen) [inline examples/lessons/12/app.js]: Comment looks wrong, no update?

File: examples/lessons/12/app.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+      uVMatrix,
+      uPMatrix: new Matrix4().perspective({fov: 45 * Math.PI / 180, aspect, near: 0.1, far: 100})
+    });
+
+    cube.render({
+      uMMatrix: uCubeMMatrix,
+      uVMatrix,
+      uPMatrix: new Matrix4().perspective({fov: 45 * Math.PI / 180, aspect, near: 0.1, far: 100})
+    });
+    animate(timeline, moonRotation, cubeRotation);
```
Comment (ibgreen) [inline examples/lessons/12/app.js]: Nit: Insert empty line before `animate`?

File: examples/lessons/12/app.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+  <p>
+    <a href="http://learningwebgl.com/blog/?p=1359" target="_blank">
+    Point lighting
+    </a>
+  <p>
+    The classic WebGL Lessons in luma.gl
+    `;
+};
+
+function degToRad(degree) {
```
Comment (ibgreen) [inline examples/lessons/12/app.js]: math.gl exports a function `radians` 

`const rad = radians(degrees)`

File: examples/lessons/12/app.js (hunk 1 above)
Comment (jianhuang01) [reply examples/lessons/12/app.js]: Good catch, thanks! 👍

File: examples/lessons/12/app.js (hunk 3 above)
Comment (jianhuang01) [reply examples/lessons/12/app.js]: Done.

File: examples/lessons/12/app.js (hunk 4 above)
Comment (jianhuang01) [reply examples/lessons/12/app.js]: Good to know. Done.

**Pull request merged**